lib/collection/src/shards/replica_set/partial_snapshot_meta.rs

```diff
@@ -1,29 +1,38 @@
 use std::sync::Arc;
+use std::sync::atomic::{self, AtomicU64, AtomicUsize};
+use std::time::{Duration, SystemTime};
 
 use crate::operations::types::{CollectionError, CollectionResult};
 
 #[derive(Debug)]
 pub struct PartialSnapshotMeta {
+    ongoing_create_snapshot_requests_tracker: RequestTracker,
```

Comment:
Feels like an overkill. I would rather make state boolean and use single read lock as a source of truth

Reply:
Discussed on call.

TL;DR: This tracks **create** partial snapshot requests (*not* recover or recover-from), and there could be multiple ongoing create partial snapshot requests at the same time (there are no locks in create snapshot path), so we track with a counter. In practice, this field will be used by cluster-manager to ensure we only create 1 (or other configured number?) partial snapshot at a time, but it still have to be a *counter* (not a bool) unless we add a lock to create partial snapshot API as well.